src/core/src/energy/functions.rs

```diff
@@ -484,20 +484,117 @@ impl Potential for Gaussian {
-            - self.a * sqrt(PI) * erfc(sqrt(self.b) * rc) / (4.0 * self.b.powf(3.0 / 2.0))
+        self.energy(rc) * rc / (2.0 * self.b) -
+        self.a * sqrt(PI) * erfc(sqrt(self.b) * rc) / (4.0 * self.b.powf(3.0 / 2.0))
     }
 
     fn tail_virial(&self, rc: f64) -> f64 {
-        3.0 * sqrt(PI) * self.a * erfc(sqrt(self.b) * rc) / (4.0 * self.b.powf(3.0 / 2.0))
-            - self.energy(rc) * rc * (2.0 * self.b * rc * rc + 3.0) / (2.0 * self.b)
+        3.0 * sqrt(PI) * self.a * erfc(sqrt(self.b) * rc) / (4.0 * self.b.powf(3.0 / 2.0)) -
+        self.energy(rc) * rc * (2.0 * self.b * rc * rc + 3.0) / (2.0 * self.b)
     }
 }
 
+/// Mie potential.
+///
+/// The following expression of the Mie potential is used:
+/// `V(r) = p * ((sigma/r)^n - (sigma/r)^m)`
+/// where `p` is a prefactor computed as
+/// `k = n / (n - m) * (n / m)^(m / (n - m)) * epsilon`,
+/// `epsilon` is an energetic constant, `sigma` is a distance constant,
+/// and `n`, `m` are the repulsive and attractive exponents, respectively.
+///
+/// # Note
+///
+/// `n` has to be larger than `m`
+/// For `m` smaller than 3.0, there is no analytic tail correction.
+///
+/// # Examples
+///
+/// ```
+/// use lumol_core::energy::Potential;
+/// use lumol_core::energy::Mie;
+///
+/// let potential = Mie::new(2.0, 10.0, 12.0, 6.0);
+/// assert_eq!(potential.energy(2.0), 0.0);
+/// assert!(f64::abs(potential.energy(3.0) + 3.203365942785746) < 1e-8);
+///
+/// assert_eq!(potential.force(2.0), 120.0);
+/// ```
+#[derive(Clone, Copy)]
+pub struct Mie {
+    /// Distance constant
+    sigma: f64,
+    /// Exponent of repulsive contribution
+    n: f64,
+    /// Exponent of attractive contribution
+    m: f64,
+    /// Energetic prefactor computed from the exponents and epsilon
+    prefac: f64,
+}
+
+impl Mie {
+    /// Return Mie potential.
+    pub fn new(sigma: f64, epsilon: f64, n: f64, m: f64) -> Mie {
+        if m >= n {
+            panic!("The repulsive exponent n has to be larger than the attractive exponent m")
+        };
+        let prefac = n / (n - m) * (n / m).powf(m / (n - m)) * epsilon;
+        Mie {
+            sigma: sigma,
+            n: n,
+            m: m,
+            prefac: prefac,
+        }
+    }
+}
+
+impl Potential for Mie {
+    fn energy(&self, r: f64) -> f64 {
+        let sigma_r = self.sigma / r;
+        let repulsive = f64::powf(sigma_r, self.n);
+        let attractive = f64::powf(sigma_r, self.m);
+        self.prefac * (repulsive - attractive)
+    }
+
+    fn force(&self, r: f64) -> f64 {
+        let sigma_r = self.sigma / r;
+        let repulsive = f64::powf(sigma_r, self.n);
+        let attractive = f64::powf(sigma_r, self.m);
+        self.prefac * (self.n * repulsive - self.m * attractive) / r
+    }
+}
+
+impl PairPotential for Mie {
+    fn tail_energy(&self, cutoff: f64) -> f64 {
+        if self.m <= 3.0 {
+            panic!("No tail correction possible for Mie potential with an exponent lower than 3.0")
```

Comment:
Should we return 0 instead here to match the behaviour on other potentials? Maybe with a warning about tail correction.

Reply:
The warning can also be emitted while parsing the input instead of here